jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/UriCompliance.java

```diff
@@ -170,6 +170,14 @@ public String getDescription()
 
     private static final AtomicInteger __custom = new AtomicInteger();
     private static final List<UriCompliance> KNOWN_MODES = List.of(DEFAULT, LEGACY, RFC3986, UNAMBIGUOUS, UNSAFE);
+    private static final EnumSet<Violation> NO_VIOLATION = EnumSet.noneOf(Violation.class);
+
+    // This saves a AccessController.executePrivileged() call caused by
```

Comment:
`executePrivileged` -> `doPrivileged`.

Reply:
I'm going to make the constant a `Set<Violation>` as then I can make it immutable and we can avoid even the simple clone operation.